forward-proxy/src/handler/mod.rs

```diff
@@ -58,7 +52,17 @@ impl ForwardHandler {
         let client = Client::new();
 
         return match client
```

Comment:
This is a really large return encapsulating a lot of code, we could refactor it down for better readability